### DISCUSSION TITLE
why i create a object which is extends container didn't show at list

### DISCUSSION THREAD

**Discussion by 1506353029** [fyne-io/fyne#4329] [Q&A] [answered]
type FunBtn struct {
	fyne.Container
	Index   int
	AddBtn  *widget.Button
	DescBtn *widget.Button
	Label   *widget.Label
	OnClick func(index int, CIndex int)
}

func NewFunBtn() *FunBtn {
	btn := &FunBtn{}
	btn.Layout = layout.NewHBoxLayout()
	btn.Label = widget.NewLabel("")
	btn.DescBtn = widget.NewButtonWithIcon("", theme.QuestionIcon(), func() { btn.onClick(0) })
	btn.AddBtn = widget.NewButtonWithIcon("", theme.ContentAddIcon(), func() { btn.onClick(1) })
	//btn.Objects = append(btn.Objects, btn.Label, btn.DescBtn, btn.AddBtn)
	btn.Add(btn.Label)
	btn.Add(btn.DescBtn)
	btn.Add(btn.AddBtn)
	btn.Index = -1
	return btn
}

	w2 := logic.CurApp.NewWindow("test")
	tlist := widget.NewList(func() int {
		return 10
	}, func() fyne.CanvasObject {
		return NewFunBtn()
	},
		func(lii widget.ListItemID, co fyne.CanvasObject) {})
	w2.SetContent(tlist)
	w2.Show()

there is nothing at list
but if i change to "return btn.Container"，it can display nomally

**Comment by 1506353029** [comment]
type BtnCard struct {
	widget.Card
	Index int
	Queue int
	init  bool
}

func NewBtnCard() *BtnCard {
	card := &BtnCard{}
	card.SetTitle(" ")
	card.SetSubTitle(" ")
	card.SetContent(container.NewHBox(
		widget.NewButtonWithIcon("单独执行", theme.MediaPlayIcon(), func() { card.onClick(5) }),
		widget.NewButtonWithIcon("从此执行", theme.MediaPlayIcon(), func() { card.onClick(6) }),
		widget.NewButtonWithIcon("命名", theme.InfoIcon(), func() { card.onClick(0) }),
		widget.NewButtonWithIcon("修改", theme.DocumentCreateIcon(), func() { card.onClick(1) }),
		widget.NewButtonWithIcon("删除", theme.DeleteIcon(), func() { card.onClick(2) }),
		widget.NewButtonWithIcon("上移", theme.MoveUpIcon(), func() { card.onClick(3) }),
		widget.NewButtonWithIcon("下移", theme.MoveDownIcon(), func() { card.onClick(4) }),
	))
	card.ExtendBaseWidget(card)
	return card
}

and i create a object which is extends card also can display normally

**Comment by andydotxyz** [comment, marked as answer]
Container cannot be extended - only Widgets can.
fyne.Container and all objects in canvas package are concrete types which the drivers implement directly.
